sklearn/decomposition/_dict_learning.py

```diff
@@ -1692,7 +1692,7 @@ class DictionaryLearning(_BaseSparseCoding, BaseEstimator):
     >>> from sklearn.datasets import make_sparse_coded_signal
     >>> from sklearn.decomposition import DictionaryLearning
     >>> X, dictionary, code = make_sparse_coded_signal(
-    ...     n_samples=100, n_components=15, n_features=20, n_nonzero_coefs=10,
+    ...     n_samples=30, n_components=15, n_features=20, n_nonzero_coefs=10,
```

Comment:
curious, how much speedup are we getting here? are we THAT slow? :grin:

Reply:
On the macOS CI build, this test is sometimes one of the slowest and can last more than 20s on an overloaded CI runner.

Locally it's fast but decreasing this makes it ~3x faster.

Reply:
I updated the description of the PR to show the impact on the macOS CI between `main` and this PR.

I observed in the latest run than the docstring of `MiniBatchDictionaryLearning` has a similar problem (and a bug in the direction on the assertion about the sparsity of the transformed data). I pushed a96dee2 to fix it and improve the speed. We will have to wait for the end of this CI run to see the concrete impact of that last commit.

Reply:
I updated the description to take a96dee2 into account.